src/snapshot.js

```diff
@@ -93,6 +110,15 @@ export function restorePlayerSnapshot(player, snapshotObject) {
 
   const suppressedRemoteTracks = snapshotObject.suppressedRemoteTracks;
   const suppressedTracks = snapshotObject.suppressedTracks;
+
+  // ensures that the iOS TextTrackList 'change' listener added in getPlayerSnapshot()
+  // doesn't persist into main content
+  if (videojs.browser.IS_IOS) {
+    const tracks = player.textTracks();
+
+    tracks.off('change');
```

Comment:
I realized that by declaring the handler outside of both `getPlayerSnapshot()` and `restorePlayerSnapshot()`, it no longer has access to the `player` unless I pass it in as an argument. But since `.on()` and `.off()` expect to be passed a function reference, I'm not sure what the best practice is for passing additional parameters. What do you recommend?